website_mass_mailing_name/static/src/js/website_mass_mailing_name.js

```diff
@@ -18,16 +18,10 @@ odoo.define("website_mass_mailing_name.subscribe", function (require) {
             return this._super(editable_mode);
         },
 
-        on_click: function() {
-            var email_error = !this.$email.val().match(/.+@.+/),
-                name_error = this.$name.length && !this.$name.val(),
-                values = {
-                    "list_id": this.$target.data('list-id'),
-                    "email": this.$email.val(),
-                };
-            // Stop on error
-            if (email_error || name_error) {
-                this.$target.addClass("has-error")
+        _onClick: function() {
+            var email_valid = this.$email[0].reportValidity(),
```

Comment:
I still think we're checking this twice. Anyway, not blocking.

Reply:
Yes, but there's a reason for that.

If we do it like it is now, UX is way better. The user gets a clear bubble indicating where the problem is. Maybe it's a wrong email, or maybe it's an empty field.

If we rely on upstream validation, it would just set all red (in fact, both fields) and say nothing. It's OK if there's only one field, but now that there are 2, if they are validated in different ways, it gets inconsistent.

The way odoo does it supports very old browsers, but has the problems explained in https://github.com/odoo/odoo/issues/14942.

However it's true that you can add a comment explaining why we're duplicating this validation here, @victormmtorres.

Reply:
Add the comment and I'll merge